src/Security/Certificate.cs

```diff
@@ -543,6 +543,38 @@ public static SecStatusCode GenerateKeyPair (NSDictionary parameters, out SecKey
 				publicKey = privateKey = null;
 			return res;
 		}
+
+#if MONOMAC
+		public static SecStatusCode GenerateKeyPair (SecKeyType type, int keySizeInBits, SecPublicPrivateKeyAttrs publicPrivateKeyAttrs, out SecKey publicKey, out SecKey privateKey)
+		{
+			if (type == SecKeyType.Invalid)
+				throw new ArgumentException ("invalid 'SecKeyType'", nameof (type));
+
+			NSMutableDictionary dic;
+			if (publicPrivateKeyAttrs != null)
+				dic = new NSMutableDictionary (publicPrivateKeyAttrs.GetDictionary ());
+			else
+				dic = new NSMutableDictionary ();
+			dic.LowlevelSetObject (type.GetConstant (), SecAttributeKey.Type);
+			dic.LowlevelSetObject (new NSNumber (keySizeInBits), SecAttributeKey.KeySizeInBits);
+			return GenerateKeyPair (dic, out publicKey, out privateKey);
+		}
+#else
```

Comment:
That conditional makes it impossible to write code that works on both macOS and iOS (and friends).
Any reason you cannot (like we discussed in Boston) expose the macOS-only signature on iOS and just use the same strong dictionary in for both public and private (in the iOS version) ?

Reply:
I don't think it's confusing since the parameters (which might need to be renamed) makes it clear depending on the overload you choose.  

> So I thought that on iOS it'd be confusing to have 2 overloads, which one should I choose?

If you have different requirements for both keys then it's the dual one, if it's the same then the single one :)

Adding the advice on the iOS specific one is a good idea too.

The goal is to allow a single source code (and eventually binaries) to cover all OS (whenever possible). If developers uses the same strong dictionary for both public/private then we're simply asking (forcing) them to code something that's also not cross-platform.

Reply:
Agree.

For the advice I was more thinking to put it on the shared API (not the iOS only one).

Something like `[Advice ("On iOS this method applies the attributes to both public and private key. To apply different attributes to each key, use 'GenerateKeyPair (SecKeyType, int, SecPublicPrivateKeyAttrs, SecPublicPrivateKeyAttrs, out SecKey, out SecKey)' instead.")]`